Userland/Libraries/LibGL/SoftwareRasterizer.cpp

```diff
@@ -368,7 +374,11 @@ static void rasterize_triangle(const RasterizerOptions& options, Gfx::Bitmap& re
                         triangle.vertices[2].tex_coord,
                         barycentric);
 
-                    *pixel = pixel_shader(uv, vertex_color);
+                    // Calculate depth of fragment for fog
+                    float z = interpolate(triangle.vertices[0].position.z(), triangle.vertices[1].position.z(), triangle.vertices[2].position.z(), barycentric);
+                    z = options.depth_min + (options.depth_max - options.depth_min) * (z + 1) / 2;
+
+                    *pixel = pixel_shader(uv, vertex_color, z);
```

Comment:
@sunverwerth I've done this separately so as to not change the scope of `float z` in the depth test (so we aren't doing if depth testing it enabled). Unfortunately going to have to take the hit here I think.